src/core_plugins/kibana/public/visualize/listing/visualize_landing_page_table.js

```diff
@@ -162,43 +121,31 @@ export class VisualizeLandingPageTable extends React.Component {
     return this.getColumnSortOrder(TYPE_COLUMN_ID);
   }
 
-  getAreAllItemsSelected() {
-    const { selectedIds, items, currentPage } = this.state;
-    const pageOfItems = this.pager.getItemsOnPage(items, currentPage);
-    const pageOfItemIds = pageOfItems.map(item => item.id);
-    return areAllItemsSelected(selectedIds, pageOfItemIds);
+  areAllItemsSelected() {
+    const pageOfItems = this.pager.getItemsOnPage(this.state.items, this.state.currentPage);
+    return areAllItemsSelected(this.state, pageOfItems);
   }
 
-  getVisualizeColumns() {
+  renderVisualizeColumns() {
     const { selectedIds } = this.state;
     return [
-      getCheckBoxColumn(this.getAreAllItemsSelected(), selectedIds, this.onToggleItem, this.onToggleAll),
-      getTitleColumn(this.getEditUrlForItem, this.getTitleSortOrder(), this.onSortByTitle),
-      getTypeColumn(this.onSortByType, this.getTypeSortOrder()),
+      getCheckBoxColumn(this.areAllItemsSelected(), selectedIds, this.onToggleItem, this.onToggleAll),
+      getTitleColumn(this.getEditUrlForItem, this.getTitleSortOrder(), this.onSort.bind(null, TITLE_COLUMN_ID)),
+      getTypeColumn(this.onSort.bind(null, TYPE_COLUMN_ID), this.getTypeSortOrder()),
     ];
   }
 
-  onDelete = () => {
-    const { deleteItems } = this.props;
-    const { selectedIds, filter } = this.state;
-    deleteItems(selectedIds).then((didDelete) => {
-      if (didDelete) {
-        this.onFilter(filter);
-      }
-    });
-  };
-
-  getActionButtons() {
+  renderActionButtons() {
     return this.state.selectedIds.length > 0
       ? <DeleteButton onClick={this.onDelete} />
       : <CreateButtonLink href={'#' + VisualizeConstants.WIZARD_STEP_1_PAGE_PATH} />;
   }
 
-  getTableContents() {
+  renderTableContents() {
     const { isFetchingItems, currentPage, items } = this.state;
```

Comment:
Some rambling thoughts:

For the sake of the following explanation, note that I put back the namespacing and renamed the file `listing_table_state` => `listing_table_actions`.

How do you feel about having these state variable names both inside `listing_table_actions` and out here?  I feel like they could get out of sync and why @weltenwort's selectors suggestion works well.  Just now I accidentally had `return { isFetching: true }` inside `ListingTableActions`, but out here had `isFetchingItems` still. 

I can see this being an easy mistake to make during a refactor.  Say we are using `ListingTableActions` for dashboard too.  Someone makes a refactor and updates the variable name in here and `ListingTableActions` but forgets that `DashboadLandingPageTable` uses it as well and forgets to update there.

 I see a few options to help this:

1) Push getters into listing_table_actions, so this becomes`ListingTableActions.getIsFetchingItems(this.state);`.
Cons: A lot of extra text.  Perhaps over-engineered?  Could simplify this by having ListingTableActions hold a reference to `this.state`.  Then can just do listingTableActions.getIsFetchingItems().  Well, now I am going around in circles because this is pretty much just what I had in the first iteration.

2.  Create a typed state definition, `ListingTableState` so the variables are defined in a single location.  A `single source of truth` if you will, in regard to the variable name. A single location for comments, a jsdoc definition, and reusable no matter what UI framework we are using. 
Could maybe use this like so:
```
constructor() {
  this.state = new ListingTableState();
}
```
Cons: not sure how react would handle a non-native object in state.  Still going around in circles as this is also close to what I first had originally: `this.state = { tableState: new ListingTableState() }`. My first implementation violated the `prefer primitives in state` style decision we were talking about. 

3.  Implement @weltenwort's selectors. 

... we can chat more about this in our meeting in a few minutes